src/views/VenuesList/VenueTiles/index.tsx

```diff
@@ -1,16 +1,46 @@
 import { VenueTile } from "./VenueTile";
 import { Venue } from "../../../shared/types/Venue.ts";
-import { FunctionComponent } from "react";
-import { VenueTilesContainer } from "./VenueTiles.styled.tsx";
+import { ChangeEvent, FunctionComponent } from "react";
+import {
+  PaginationWrapper,
+  ShowAndSortContainer,
+  SortButton,
+  VenueTilesContainer,
+  VenueTilesWrapper,
+} from "./VenueTiles.styled.tsx";
+import { Pagination, Typography } from "@mui/material";
+import { SetURLSearchParams } from "react-router-dom";
 
 interface VenueTiles {
   venues: Venue[];
+  numberOfVenues: number | null;
+  setSearchParams: SetURLSearchParams;
 }
 
-export const VenueTiles: FunctionComponent<VenueTiles> = ({ venues }) => {
+export const VenueTiles: FunctionComponent<VenueTiles> = ({
+  venues,
+  numberOfVenues,
+  setSearchParams,
+}) => {
+  const pagesNumber = Math.ceil((numberOfVenues ?? 1) / 18);
+
+  const changePage = (_event: ChangeEvent<unknown>, value: number) =>
+    setSearchParams({ page: String(value) });
```

Comment:
Create the `useVenueTiles` hook for that. Also, make this code look better - don't hesitate to declare additional variables, occupy more lines